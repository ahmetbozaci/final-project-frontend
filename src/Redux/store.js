import { createStore, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import logger from 'redux-logger';
import { composeWithDevTools } from 'redux-devtools-extension';
import rootReducer from './rootReducer';
import fetchDoctors from './doctor/doctorActions';

const store = createStore(
  rootReducer,
  composeWithDevTools(applyMiddleware(thunk, logger)),
);
store.dispatch(fetchDoctors());

export default store;
