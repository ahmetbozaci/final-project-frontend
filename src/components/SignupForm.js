/* eslint-disable no-unused-vars */
import { useState } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { Form, Button } from 'react-bootstrap';
import { Link, useNavigate } from 'react-router-dom';
import createUser from '../Redux/Signup/API';
import './style/signup.css';

const Signup = () => {
  const status = useSelector((state) => state.signupReducer.status);
  const navigate = useNavigate();
  const dispatch = useDispatch();
  const [state, setState] = useState({
    name: '',
    email: '',
    password: '',
    passwordConfirmation: '',
  });

  const {
    name, email, password, passwordConfirmation,
  } = state;

  const handleChange = (event) => {
    const { name, value } = event.target;
    setState({ ...state, [name]: value });
  };

  const handleSubmit = async (event) => {
    event.preventDefault();
    const newUser = {
      name,
      email,
      password,
      passwordConfirmation,
    };
    dispatch(createUser(newUser));
  };

  return (
    <Form onSubmit={handleSubmit} className="form m-auto mt-5">
      <h1 className="text-center">Sign Up</h1>
      <Form.Group className="mb-2" controlId="formBasicUsername">
        <Form.Label>
          <h5 className="text-muted">Username</h5>
        </Form.Label>
        <Form.Control
          placeholder="Username"
          type="text"
          name="name"
          value={name}
          onChange={handleChange}
        />
      </Form.Group>
      <Form.Group className="mb-2" controlId="formBasicEmail">
        <Form.Label>
          <h5 className="text-muted">Email address</h5>
        </Form.Label>
        <Form.Control
          placeholder="Enter email"
          type="text"
          name="email"
          value={email}
          onChange={handleChange}
        />
      </Form.Group>
      <Form.Group className="mb-2" controlId="formBasicPassword">
        <Form.Label>
          <h5 className="text-muted">Password</h5>
        </Form.Label>
        <Form.Control
          type="password"
          placeholder="Enter password"
          name="password"
          value={password}
          onChange={handleChange}
        />
      </Form.Group>
      <Form.Group className="mb-2" controlId="formBasicPassword">
        <Form.Label>
          <h5 className="text-muted">Password confirmation</h5>
        </Form.Label>
        <Form.Control
          type="password"
          placeholder="Enter password again"
          name="passwordConfirmation"
          value={passwordConfirmation}
          onChange={handleChange}
        />
      </Form.Group>
      <div className="d-grid gap-2 mt-3">
        <Button variant="primary" size="lg" type="submit">
          Sign Up
        </Button>
      </div>
      <p className="text-muted signin-text">
        <small>
          Already registered
          <Link to="/login" className="signin-link">
            {' '}
            sign in
          </Link>
        </small>
      </p>
    </Form>
  );
};

export default Signup;
