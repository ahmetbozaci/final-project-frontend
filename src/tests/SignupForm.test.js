import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../Redux/store';
import SignupForm from '../components/SignupForm';

it('SignupForm renders correctly', () => {
  const tree = renderer
    .create(<Provider store={store}><Router><SignupForm /></Router></Provider>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
