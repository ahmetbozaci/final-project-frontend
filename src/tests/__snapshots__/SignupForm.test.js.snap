// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignupForm renders correctly 1`] = `
<div
  className="form-container p-5"
>
  <form
    className="signUp-form m-auto py-3"
    onSubmit={[Function]}
  >
    <h1
      className="text-center"
    >
      Sign Up
    </h1>
    <div
      className="mb-2"
    >
      <label
        className="form-label"
        htmlFor="formBasicUsername"
      >
        <h5
          className="text-muted"
        >
          Username
        </h5>
      </label>
      <input
        className="form-control"
        id="formBasicUsername"
        name="name"
        onChange={[Function]}
        placeholder="Username"
        required={true}
        type="text"
        value=""
      />
    </div>
    <div
      className="mb-2"
    >
      <label
        className="form-label"
        htmlFor="formBasicEmail"
      >
        <h5
          className="text-muted"
        >
          Email address
        </h5>
      </label>
      <input
        className="form-control"
        id="formBasicEmail"
        name="email"
        onChange={[Function]}
        placeholder="Enter email"
        required={true}
        type="text"
        value=""
      />
    </div>
    <div
      className="mb-2"
    >
      <label
        className="form-label"
        htmlFor="formBasicPassword"
      >
        <h5
          className="text-muted"
        >
          Password
        </h5>
      </label>
      <input
        className="form-control"
        id="formBasicPassword"
        name="password"
        onChange={[Function]}
        placeholder="Enter password"
        required={true}
        type="password"
        value=""
      />
    </div>
    <div
      className="mb-2"
    >
      <label
        className="form-label"
        htmlFor="formBasicPassword"
      >
        <h5
          className="text-muted"
        >
          Password confirmation
        </h5>
      </label>
      <input
        className="form-control"
        id="formBasicPassword"
        name="passwordConfirmation"
        onChange={[Function]}
        placeholder="Enter password again"
        required={true}
        type="password"
        value=""
      />
      <small
        className="text-muted form-text"
        id="passwordError"
      />
    </div>
    <div
      className="text-center"
    >
      <button
        className="btn btn-btn btn-primary btn-lg"
        disabled={false}
        type="submit"
      >
        Sign Up
      </button>
    </div>
    <p
      className="text-muted text-center signin-text"
    >
      <small>
        Already registered
        <a
          className="signin-link"
          href="/login"
          onClick={[Function]}
        >
           
          sign in
        </a>
      </small>
    </p>
  </form>
</div>
`;
